components/central-application-gateway/internal/proxy/factory.go

```diff
@@ -41,7 +42,10 @@ func New(
 		gwURL := *u
 		gwURL.Path = "/" + strings.Join(split[:2], "/")
 
-		return apiIdentifier, targetAPIPath, &gwURL, nil
+		// TODO: make sure
+		targetURL, _ := url.Parse(targetAPIPath) // it shouldn't be possible for this to fail?
```

Comment:
Is it done?

Reply:
I'd much rather handle the potential error, but I am not sure how it'd happen...
```go
if err != nil {
    return model.APIIdentifier{}, nil, nil, apperrors.WrongInput("Path has to be a valid path")
}
```